### ISSUE TITLE
systemd-boot now defaults `/efi` as the folder for initramfs images instead of `/boot`. How does booster cope with that?

### ISSUE THREAD

**Issue by enihcam** [anatol/booster#246]
I'm asking this question because I see some hardcoded `/boot` in booster codebase.

**Comment by enihcam** [comment]
As you can see, [dracut](https://github.com/dracutdevs/dracut) fixed this already, so maybe you can take the following code as a reference?
https://github.com/dracutdevs/dracut/commit/093bc9b5cfc071409a20351a669c594c748eb647
https://github.com/dracutdevs/dracut/commit/49ea6c42db7180eec5ba57e082a38d116f2d17a5

**Comment by enihcam** [comment]
Instead of ad-hoc script, I thought an elegant way to copy initramfs artifacts over to $BOOT directory is to use [systemd-kernel-install](https://www.freedesktop.org/software/systemd/man/kernel-install.html)

An example would be:
https://gitlab.com/dalto.8/kernel-install-for-dracut

What do you think?

**Comment by c3Ls1US** [comment]
On Arch Linux, Booster provides a shell script that puts the initramfs image at /boot. The shell script does not currently accommodate this causing the image to not be built. As a result, since there's no image, the bootloader probably wouldn't find it.

As a workaround, you would have to manually build and specify the appropriate path

**Comment by enihcam** [comment]
> On Arch Linux, Booster provides a shell script that puts the initramfs image at /boot. The shell script does not currently accommodate this which (should) cause the image to not be built. As a result, since there's no image, this implies that the bootloader wouldn't find it.
> 
> As a workaround, you would have to manually build and specify the appropriate path

right, that's the problem. as a workaround, i manually replaced `/boot` with `/efi`. hopefully there'll be a more elegant approach to resolve this.

**Comment by c3Ls1US** [comment]
@enihcam To follow up, is there any specific reason why you do not have a `/boot` directory? I understand the case where one can have both `/boot` and `/efi` but not only `/efi`.

This is an Arch Linux script so the default boot path for initramfs images and other components is `/boot`.  If going by mkinitcpio and assuming you have only 1 ESP partition, `/efi` seems to be only reserved for UKIs which (should) require the user to adjust the new mount point of the ESP to `/efi`, leaving boot files residing within the root partition.

[I understand that there are cases for other ESP mount points](https://wiki.archlinux.org/title/EFI_system_partition#Typical_mount_points), but what's yours specifically?

**Comment by kjkent** [comment]
I cannot remember which, either grub, arch's mkinitcpio, or dracut (but I think one of the first two) has some functionality hard-coded to `/boot`. So I went with that, but I've found that even creating `/efi` causes systemd to enable `efi.automount`, which does as it's named. It didn't cause issues on my machine but could others. I can't remember what created `/efi`, but I removed it and everything works fine.

This is outside the scope of the issue, but relevant -- @anatol , is compatibility with systemd's `kernel-install` & `systemd-ukify` something that you'd want? If so, I have written a plugin script for booster to work in this context & could send a PR for you to look at. No worries if not!

**Comment by lpnh** [comment]
I understand using `/boot` as the default path, but I think it would be nice to have a solution that doesn't require manually editing the scripts.

**Comment by notritsuwu** [comment]
I ran into this problem as well with ESP mount point at /efi/ with systemd-boot, and the systemd-ukify solution doesn't work for me as well (it can build uki successfully, but still uses the wrong directory too), I'm sure this issue really needs a actual fix instead of just messing with the built in rebuild script.
